### ISSUE TITLE
Lint/EmptyConditionalBody nil dereference when nested in block and preceding code

### ISSUE THREAD

**Issue by BrianHawley** [rubocop/rubocop#10538]
I ran into this with my own code, but came up with a minimized reproduction for the issue. Disabling the cop with pragma comments doesn't help, as we get the same error, plus an additional one saying the cop was disabled unnecessarily.

--------

## Expected behavior

I expected the Lint/EmptyConditionalBody cop to report normally.

## Actual behavior

I got this result (with personal details redacted):
```
Scanning blah.rb
An error occurred while Lint/EmptyConditionalBody cop was inspecting blah.rb:5:2.
undefined method `line' for nil:NilClass
/.../gems/rubocop-1.27.0/lib/rubocop/cop/mixin/comments_help.rb:55:in `find_end_line'
/.../gems/rubocop-1.27.0/lib/rubocop/cop/mixin/comments_help.rb:16:in `contains_comments?'
/.../gems/rubocop-1.27.0/lib/rubocop/cop/lint/empty_conditional_body.rb:62:in `on_if'
/.../gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:100:in `public_send'
/.../gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:100:in `block (2 levels) in trigger_responding_cops'
/.../gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
/.../gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:99:in `block in trigger_responding_cops'
/.../gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:98:in `each'
/.../gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:98:in `trigger_responding_cops'
/.../gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:69:in `on_if'
/.../gems/rubocop-ast-1.17.0/lib/rubocop/ast/traversal.rb:136:in `block in on_dstr'
/.../gems/rubocop-ast-1.17.0/lib/rubocop/ast/traversal.rb:136:in `each'
/.../gems/rubocop-ast-1.17.0/lib/rubocop/ast/traversal.rb:136:in `on_dstr'
/.../gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/.../gems/rubocop-ast-1.17.0/lib/rubocop/ast/traversal.rb:157:in `on_block'
/.../gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/.../gems/rubocop-ast-1.17.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/.../gems/rubocop-1.27.0/lib/rubocop/cop/commissioner.rb:86:in `investigate'
/.../gems/rubocop-1.27.0/lib/rubocop/cop/team.rb:155:in `investigate_partial'
/.../gems/rubocop-1.27.0/lib/rubocop/cop/team.rb:83:in `investigate'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:309:in `inspect_file'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:253:in `block in do_inspection_loop'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:287:in `block in iterate_until_no_changes'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:280:in `loop'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:280:in `iterate_until_no_changes'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:249:in `do_inspection_loop'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:130:in `block in file_offenses'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:155:in `file_offense_cache'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:129:in `file_offenses'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:120:in `process_file'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:101:in `block in each_inspected_file'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:100:in `each'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:100:in `reduce'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:100:in `each_inspected_file'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:86:in `inspect_files'
/.../gems/rubocop-1.27.0/lib/rubocop/runner.rb:47:in `run'
/.../gems/rubocop-1.27.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/.../gems/rubocop-1.27.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/.../gems/rubocop-1.27.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/.../gems/rubocop-1.27.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/.../gems/rubocop-1.27.0/lib/rubocop/cli/command.rb:11:in `run'
/.../gems/rubocop-1.27.0/lib/rubocop/cli/environment.rb:18:in `run'
/.../gems/rubocop-1.27.0/lib/rubocop/cli.rb:71:in `run_command'
/.../gems/rubocop-1.27.0/lib/rubocop/cli.rb:78:in `execute_runners'
/.../gems/rubocop-1.27.0/lib/rubocop/cli.rb:47:in `run'
/.../gems/rubocop-1.27.0/exe/rubocop:12:in `block in <top (required)>'
/.../lib/ruby/3.0.0/benchmark.rb:308:in `realtime'
/.../gems/rubocop-1.27.0/exe/rubocop:12:in `<top (required)>'
/.../bin/rubocop:25:in `load'
/.../bin/rubocop:25:in `<main>'
/.../bin/ruby_executable_hooks:22:in `eval'
/.../bin/ruby_executable_hooks:22:in `<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Lint/EmptyConditionalBody cop was inspecting blah.rb:5:2.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.27.0 (using Parser 3.1.2.0, rubocop-ast 1.17.0, running on ruby 3.0.3 x86_64-darwin19)
Finished in 1.2193400000105612 seconds
```

## Steps to reproduce the problem

Using rubocop 1.27.0, check a file with this code:
```ruby
# frozen_string_literal: true

blah do
  something
  if condition
  end
end
```

The frozen string literal stuff doesn't matter; it's just required by my local setup.

This code, with a comment in the condition block, will also trigger the problem:
```ruby
blah do
  something
  if condition
    # comment
  end
end
```

## RuboCop version

```
1.27.0 (using Parser 3.1.2.0, rubocop-ast 1.17.0, running on ruby 3.0.3 x86_64-darwin19)
  - rubocop-rails 2.14.2
  - rubocop-rspec 2.9.0
```

**Comment by BrianHawley** [comment]
The problem was introduced in #10491.

**Comment by koic** [comment]
This issue is a dup with #10514 and resolved by #10516.
